File: medilink_admin/lib/models/rating.dart

class Rating {
   String? id;
   String? healthcareProvider;
   String? patient;
  double? rating;
  String? review;

  Rating({
    this.id,
    this.healthcareProvider,
    this.patient,
    this.rating,
    this.review,
  });

  factory Rating.fromJson(Map<String, dynamic> json) {
    return Rating(
      id: json['_id'] as String?,
      healthcareProvider: json['healthcareProvider']  as String?,
      patient:
          json['patient']  as String?,
      rating: json['rating']?.toDouble(),
      review: json['review'] as String?,
    );
  }
}

File: back_end/models/rating.js
const mongoose = require('mongoose');

const ratingSchema = new mongoose.Schema({
  healthcareProvider: { type: mongoose.Schema.Types.ObjectId, ref: 'HealthcareProvider' },
  patient: { type: mongoose.Schema.Types.ObjectId, ref: 'Patient' },
  rating: { type: Number, required: true },
  review: String,
});

const Rating = mongoose.model('Rating', ratingSchema);

module.exports = Rating;
